rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -418,20 +422,34 @@ ImageData ROSImageTexture::setFormatAndNormalizeDataIfNecessary(ImageData image_
     image_data.encoding_ == sensor_msgs::image_encodings::MONO16)
   {
     image_data.size_ /= sizeof(uint16_t);
-    std::vector<uint8_t> buffer = normalize<uint16_t>(
+    if (!bufferptr_) {
+      bufferptr_ = std::make_shared<std::vector<uint8_t>>(image_data.size_);
+    } else if (static_cast<size_t>(bufferptr_->size()) != image_data.size_) {
+      bufferptr_->resize(image_data.size_, 0);
+    }
```

Comment:
A few different things about this code:

1. I feel like we can simplify this idiom by allocating at least one byte to `bufferptr_` in the constructor.  Then all of the places we do this, we can change this to:
```
if (bufferptr_->size() != image_data.size_) {
  bufferptr_->resize(image_data.size_, 0);
}
```
2. I don't think we need the static_cast to size_t; both `bufferptr_->size()` and `image_data.size_` are `size_t`.
3. Technically, we only need to resize the buffer if it is *smaller* than what we currently need.  That could save some reallocation, at the expense of some wasted memory.  I don't feel strongly about this point, though.
4. I can't quite put my finger on it, but the resize logic looks incorrect.  In particular, just prior to this code, we set `image_data.size_` to the number of uint16_t elements in the incoming piece of data.  But then we allocate that number of *uint8_t* bytes, which seems like we wouldn't have enough data in the bufferptr.  For example, assume that we have 10 uint16_t elements.  In that case, we'll allocate 10 bytes to the bufferptr, not 20.  It feels like we can overrun this when we are normalizing, but I'm not sure.

Reply:
I think these two lines are calculating the uint_8 size of the buffer.

```cpp
image_data.size_ /= sizeof(uint16_t);
...
image_data.size_ /= sizeof(float);
```

Reply:
I can't see the reason to divide here. From looking at the origin of the data, we'd just want to allocate `image_data.size_`? I believe it's already the number of bytes not the number of elements? 

https://github.com/ros2/rviz/blob/018de8f730f3d9857395b7345acb5b8c55f38656/rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp#L235

https://github.com/ros2/common_interfaces/blob/a3a0dde2ba184b01cdc59a3003728906de3240a9/sensor_msgs/msg/Image.msg#L26